src/installer.ts

```diff
@@ -233,14 +252,17 @@ export async function installKubectl(context: vscode.ExtensionContext): Promise<
 
 	extension.shareMessageInMainOutputChannel(`Downloading Kubernetes cli tool from ${kubectlUrl} to ${downloadFile}`);
 
-	const downloadResult = await downloader.to(kubectlUrl, downloadFile);
-	if (failed(downloadResult)) {
-		return { succeeded: false, error: [`Failed to download kubectl: ${downloadResult.error[0]}`] };
-	}
+	await downloadAndExtract(kubectlUrl, binFile, installFolder, true)
+	.then( async (flag) => {
+		console.log(`Downloaded ${downloadFile} successfully: ${flag}`);
+	}).catch ( (error) => {
+		console.log(error);
+		return { succeeded: false, error: [`Failed to download kubectl: ${error}`] };
+	});
 	await config.addKubectlPathToConfig(downloadFile);
 
 	if (shell.isUnix()) {
-		fs.chmodSync(downloadFile, '0777');
+		fs.chmodSync(downloadFile, '0700');
```

Comment:
why do you need to change rights when changing the download approach?
If giving a 700, won't it means that the user needs to be an admin to be able to launch it? or that he/she launched VS Code in admin mode?

Reply:
Nope....it means OWNER/USER is allowed everything, GROUP is allowed nothing, OTHERS is allowed nothing. So that change basically improves the security because it doesn't allow other users or groups to access the users home directory where this file is stored.

Reply:
ah ok, seems better effectively.
remains my question:

why do you need to change rights when changing the download approach?

Reply:
Why can't u ask the question you actually want to ask? "Can you make this change in a separate PR?" ;)

Reply:
because if the answer is that we need this right modification to have the download progress working, it should not be in another PR. I don't think so but I'm not sure so I'm asking.

Reply:
@lhein helped change from 0777 to 0700 a while back for the kamel CLI and I just never changed it for the kubernetes side. Since we are working in this code, it did not seem a stretch to change it at this time. I honestly don't see the need to break this out

Reply:
advantages:
- smaller PR
- this modification will no tbe blocked by user comments on the progress change
- allows to have a better tracing of modifications in history
- easier to think to add an entry in changelog
- easier to understand that the modification of the permission is not needed for the progress
drawbacks:
- extra branch and PR to create

Reply:
@apupier it's one line in the context of another change that still affects the same code. it's staying put.